internal/pkg/manifest/validate_env.go

```diff
@@ -41,6 +42,17 @@ func (e EnvironmentConfig) validate() error {
 	if e.IsIngressRestrictedToCDN() && !e.CDNConfig.CDNEnabled() {
 		return errors.New("CDN must be enabled to limit security group ingress to CloudFront")
 	}
+	if e.CDNConfig.CDNEnabled() {
+		if e.HTTPConfig.Public.Certificates == nil {
+			if e.CDNConfig.CDNConfig.Certificate != nil {
+				return errors.New("must import ALB certificates to set \"cdn.certificate\"")
+			}
+		} else {
+			if e.CDNConfig.CDNConfig.Certificate == nil {
+				return errors.New("must set \"cdn.certificate\" to import ALB certificates with CDN enabled")
+			}
+		}
+	}
```

Comment:
This condition has multiple nested ifs, I was thinking what we can do with it. I tried doing few permutations with the if logic and it passes your tests in validate_env but fails few other tests for some weird reasons. 
We can at least take this `e.CDNConfig.CDNConfig.Certificate` out in a variable and make it more readable.